pkg/splunk/enterprise/util.go

```diff
@@ -1172,6 +1129,31 @@ func setupAppInitContainers(client splcommon.ControllerClient, cr splcommon.Meta
 	}
 }
 
+// isAppAlreadyDownloaded checks if the app is already present on the operator pod
+func isAppAlreadyDownloaded(downloadWorker *PipelineWorker) bool {
+	scopedLog := log.WithName("isAppAlreadyDownloaded").WithValues("app name", downloadWorker.appDeployInfo.AppName)
+
+	scope := getAppSrcScope(downloadWorker.afwConfig, downloadWorker.appSrcName)
+	kind := downloadWorker.cr.GetObjectKind().GroupVersionKind().Kind
+
+	localPath := filepath.Join(splcommon.AppDownloadVolume, "downloadedApps", downloadWorker.cr.GetNamespace(), kind, downloadWorker.cr.GetName(), scope, downloadWorker.appSrcName) + "/"
+	localAppFileName := getLocalAppFileName(localPath, downloadWorker.appDeployInfo.AppName, downloadWorker.appDeployInfo.ObjectHash)
+
+	// check if the app is present on operator pod
+	file, err := os.Open(localAppFileName)
+
+	if os.IsNotExist(err) {
+		scopedLog.Info("App not present on operator pod")
+		return false
+	}
+
+	// update the state to be download complete
+	updatePplnWorkerPhaseInfo(downloadWorker.appDeployInfo, enterpriseApi.PhaseDownload, 0, enterpriseApi.AppPkgDownloadComplete)
```

Comment:
Nit: Can we do this step at the calling place of this function, so that this function can be re-used across the phases?